bundler/lib/bundler/templates/newgem/lib/newgem.rb.tt

```diff
@@ -4,6 +4,12 @@ require_relative "<%= File.basename(config[:namespaced_path]) %>/version"
 <%- if config[:ext] -%>
 require_relative "<%= File.basename(config[:namespaced_path]) %>/<%= config[:underscored_name] %>"
 <%- end -%>
+<%- if config[:zeitwerk] -%>
+
+require "zeitwerk"
+loader = Zeitwerk::Loader.for_gem
+loader.setup
+<%- end -%>
```

Comment:
Nice! For normal gems it would look like this:

```ruby
require "zeitwerk"
loader = Zeitwerk::Loader.for_gem
loader.setup
```

For extension-like gems, according to https://github.com/fxn/zeitwerk#for_gem_extension, it should look like this for the example of a gem named `net-http-niche_feature`:

```ruby
require "zeitwerk"
loader = Zeitwerk::Loader.for_gem_extension(Net::Http)
loader.setup
```

Here is the first draft of something that would work along those lines:

```suggestion
<%- if config[:zeitwerk] -%>

require "zeitwerk"
  <%- if config[:name].include?("-") -%>
loader = Zeitwerk::Loader.for_gem_extension(<%= config[:constant_array][0..-2].join("::") %>)
  <%- else -%>
loader = Zeitwerk::Loader.for_gem
  <%- end -%>
loader.setup
<%- end -%>
```

(I had to do something similar for a custom private gem generator of mine)

Reply:
Good point @jeromedalbert! I'll add that and write some tests later today 👍.